cmd/goexpander/expander/expander.go

```diff
@@ -0,0 +1,146 @@
+	for {
+		err := decoder.Decode(&document)
+		if err != nil {
+			if strings.Contains(err.Error(), "Expected document start at line") {
+				return stream, nil
+			}
+			return nil, err
+		}
+		// Now it's held in document but we have to do a bit of a dance to get it in a form that can
+		// be marshaled as JSON for our API response.  The fundamental problem is that YAML is a
+		// superset of JSON in that it can represent non-string keys, and full IEEE floating point
+		// values (NaN etc).  JSON only allows string keys and its definition of a number is based
+		// around a sequence of digits.
+
+		// Kubernetes does not make use of these features, as it uses YAML as just "pretty JSON".
+		// Consequently this does not affect Helm either.  However, both candiedyaml and go-yaml
+		// return types that are too wide for JSON marshalling (i.e. map[interface{}]interface{}
+		// instead of map[string]interface{}), so we have to do some explicit conversion.  Luckily,
+		// ghodss/yaml has code to help with this, since decoding from YAML to JSON-marshalable
+		// values is exactly the problem that it was designed to solve.
+
+		// 1) Marshal it back to YAML string.
+		yamlBytes, err := candiedyaml.Marshal(document)
+		if err != nil {
+			return nil, err
+		}
+
+		// 2) Use ghodss/yaml to unmarshal that string into JSON-compatible data structures.
+		var jsonObj interface{}
+		err = yaml.Unmarshal(yamlBytes, &jsonObj)
+		if err != nil {
+			return nil, err
+		}
+
+		// Now it's suitable for embedding in an API response.
+		stream = append(stream, jsonObj)
+	}
+}
+
+type expander struct {
+}
+
+// NewExpander returns an Go Templating expander.
+func NewExpander() expansion.Expander {
+	return &expander{}
+}
+
+// ExpandChart resolves the given files to a sequence of JSON-marshalable values.
+func (e *expander) ExpandChart(request *expansion.ServiceRequest) (*expansion.ServiceResponse, error) {
+
+	err := expansion.ValidateRequest(request)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO(dcunnin): Validate via JSONschema.
+
+	chartInv := request.ChartInvocation
+	// chartFile := request.Chart.Chartfile
+	chartMembers := request.Chart.Members
+
+	resources := []interface{}{}
+	for _, file := range chartMembers {
+		name := file.Path
+		content := file.Content
+		tmpl := template.New(name).Funcs(sprig.TxtFuncMap())
+
+		for _, otherFile := range chartMembers {
+			otherName := otherFile.Path
+			otherContent := otherFile.Content
+			if name == otherName {
+				continue
+			}
+			_, err := tmpl.Parse(string(otherContent))
+			if err != nil {
+				return nil, err
+			}
+		}
+
+		// Have to put something in that resolves non-empty or Go templates get confused.
+		_, err := tmpl.Parse("# Content begins now")
+		if err != nil {
+			return nil, err
+		}
+
+		tmpl, err = tmpl.Parse(string(content))
+		if err != nil {
+			return nil, err
+		}
+
+		generated := bytes.NewBuffer(nil)
+		err = tmpl.ExecuteTemplate(generated, name, chartInv.Properties)
```

Comment:
nit: This should be collapsed to one line: `if err := tmpl.Ex....; err != nil {`